pkg/ebpf/benchmark_test.go

```diff
@@ -0,0 +1,156 @@
+// +build linux_bpf
+
+package ebpf
+
+import (
+	"encoding/json"
+	"fmt"
+	"log"
+	"os/exec"
+	"sort"
+	"strings"
+	"sync"
+	"testing"
+	"time"
+
+	bpflib "github.com/DataDog/ebpf"
+	"github.com/stretchr/testify/require"
+)
+
+var (
+	bpfToolPath       string
+	currKernelVersion uint32
+	readVars          sync.Once
+)
+
+type bpftoolStats struct {
+	RunTimeNS float64 `json:"run_time_ns"`
+	RunCnt    float64 `json:"run_cnt"`
+
+	XlatedBytes float64 `json:"bytes_xlated"`
+	JITedBytes  float64 `json:"bytes_jited"`
+}
+
+func readGlobalVars() {
+	var err error
+	currKernelVersion, err = bpflib.CurrentKernelVersion()
+	if err != nil {
+		log.Fatalf("unable to read kernel version: %s\n", err)
+	}
+	bpfToolPath, err = exec.LookPath("bpftool")
+	if err != nil {
+		log.Fatalf("unable to find bpftool in PATH: %s\n", err)
+	}
+}
+
+func setBPFStatsCollection(b *testing.B, enable bool) error {
+	if currKernelVersion < stringToKernelCode("5.1.0") {
+		b.Skip("kernel version must be at least 5.1.0 for kernel-level eBPF stats collection")
+		return nil
+	}
+
+	flag := 0
+	if enable {
+		flag = 1
+	}
+	cmd := exec.Command("sysctl", "-w", fmt.Sprintf("kernel.bpf_stats_enabled=%d", flag))
+	if out, err := cmd.CombinedOutput(); err != nil {
+		return fmt.Errorf("error enabling bpf stats: %s %w", out, err)
```

Comment:
error message sounds like it is always trying to enable bpf stats, rather than set the boolean.  maybe "error setting bpf_stats_enabled" or something?

Reply:
Fixed. Thanks!